src/backend/distributed/deparser/deparse_role_stmts.c

```diff
@@ -359,7 +359,15 @@ AppendGrantRoleStmt(StringInfo buf, GrantRoleStmt *stmt)
 			if (strcmp(opt->defname, "admin") == 0)
 			{
 				appendStringInfo(buf, "ADMIN OPTION FOR ");
-				break;
+			}
+			else if (strcmp(opt->defname, "inherit") == 0);
```

Comment:
If you check at the beginning of this code block, you will notice that this part is actually for the REVOKE option, and not the GRANT
```C
if (!stmt->is_grant)
```
So, you need to modify this part for the REVOKE command.
You should also modify the GRANT command part, which is a bit further below in the part that starts with:
```C
if (stmt->is_grant)
```